src/sdk/namespace_fs.js

```diff
@@ -565,8 +647,20 @@ class NamespaceFS {
             }
             // TODO use file xattr to store md5_b64 xattr, etc.
             const stat = await nb_native().fs.stat(fs_account_config, upload_path);
+            if (params.xattr) {
+                let target_file;
+                try {
+                    target_file = await nb_native().fs.open(fs_account_config, upload_path, 'a', get_umasked_mode(config.BASE_MODE_FILE));
```

Comment:
I would push this logic to a function and call it from inside _upload_stream and _copy_stream since it already has the target_file open there.